src/renderer/features/balances/model/balance-subscription-model.ts

```diff
@@ -81,28 +105,39 @@ const unsubscribeBalancesFx = createEffect(async ({ subscription }: UnsubscribeP
     return;
   }
 
-  const [balanceUnsubs, lockUnsubs] = await subscription.subscription;
+  const [balanceUnsubs, lockUnsubs] = await Promise.all(subscription.subscription);
 
   balanceUnsubs.forEach((fn) => fn());
   lockUnsubs.forEach((fn) => fn());
 });
 
+const populateBalancesFx = createEffect((accounts: AccountId[]): Promise<Balance[]> => {
+  return balanceService.getBalances(accounts);
+});
+
+sample({
+  clock: walletModel.$activeAccounts,
+  fn: (accounts) => accounts.map((account) => account.accountId),
+  target: populateBalancesFx,
+});
+
+sample({
+  clock: populateBalancesFx.doneData,
+  target: balanceModel.$balances,
+});
+
 sample({
-  clock: [networkModel.$connectionStatuses, walletModel.$activeAccounts],
+  clock: throttle({
+    source: combine([networkModel.$connectionStatuses, walletModel.$activeAccounts]),
+    timeout: 500,
```

Comment:
Good to introduce a constant for `500`
What about other values 250, 750, 1000, 1500?

Reply:
tried 1000, no big difference with 500